src/middlewares/userMiddleware.ts

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import express from 'express'
+import userSchema from '../validations/userValidation'
+
+const validateRequest =
+  (schema: any) =>
+  async (
+    req: express.Request,
+    res: express.Response,
+    next: express.NextFunction,
+  ) => {
+    const { error } = schema.validate(req.body)
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message })
+    }
+    return next()
+  }
```

Comment:
```suggestion
const validateRequest =
  (schema: SchemaType, path: 'body'|'params'|'query'='body') =>
  async (
    req: express.Request,
    res: express.Response,
    next: express.NextFunction,
  ) => {
    const { error } = schema.validate(req[path])
    if (error) {
      return res.status(400).json({ message: error.details[0].message })
    }
    return next()
  }
```
this can be move to some place like utils or helpers where it can be reused and export it